velox/exec/VectorHasher.h

```diff
@@ -172,6 +172,11 @@ class VectorHasher {
   // computeValueIds(). The decoded vector can be accessed via decodedVector()
   // getter.
   void decode(const BaseVector& vector, const SelectivityVector& rows) {
+    VELOX_CHECK(
+        type_->kindEquals(vector.type()),
```

Comment:
Should we compare child of `vector.type()` here?

Reply:
@Yuhta Not sure I understand. Would you clarify?

Reply:
No, 'vector' here is a column. Here is an example usage:

```
  for (auto i = 0; i < hashers_.size(); ++i) {
    auto& hasher = hashers_[i];
    if (hasher->channel() != kConstantChannel) {
      hashers_[i]->decode(*input.childAt(hasher->channel()), rows_);
      hashers_[i]->hash(rows_, i > 0, hashes_);
    } else {
      hashers_[i]->hashPrecomputed(rows_, i > 0, hashes_);
    }
  }
```

Reply:
Then we should remove the `childAt` call in line 179 below

Reply:
Good catch. Thank you.